### ISSUE TITLE
Issue with Flyway and db pool during dev restart

### ISSUE THREAD

**Issue by dmarko484** [quarkusio/quarkus#14623]
**Describe the bug**
I'm developing with Quarkus 1.11.0.Final and having an issues with Flyway during development restarts. I'm running Hibernate with Panache, Flyway with PostgreSQL in IntelliJ using maven quarkus:dev: so restarts are working here.  Now when Flyway is integrated, then when I change a code the restart is initiated and I'm getting error regarding not available connection pool:
```
ava.sql.SQLException: This pool is closed and does not handle any more connections!
	at io.agroal.pool.Poolless.beforeAcquire(Poolless.java:154)
	at io.agroal.pool.Poolless.getConnection(Poolless.java:160)
	at io.agroal.pool.DataSource.getConnection(DataSource.java:81)
	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:59)
	at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:69)
	at org.flywaydb.core.Flyway.execute(Flyway.java:507)
	at org.flywaydb.core.Flyway.migrate(Flyway.java:165)
	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:61)
	at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy_0(FlywayProcessor$createBeansAndStartActions-1520831253.zig:80)
	at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy(FlywayProcessor$createBeansAndStartActions-1520831253.zig:40)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:753)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:97)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
	at java.base/java.lang.Thread.run(Thread.java:834)
Resulted in: org.flywaydb.core.internal.exception.FlywaySqlException: 
Unable to obtain connection from database: This pool is closed and does not handle any more connections!
--------------------------------------------------------------------------------------------------------
SQL State  : null
Error Code : 0
Message    : This pool is closed and does not handle any more connections!

	at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:68)
	... 19 more
Resulted in: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:940)
	... 12 more
Resulted in: java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.dev.appstate.ApplicationStateNotification.waitForApplicationStart(ApplicationStateNotification.java:51)
	at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:145)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:190)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:174)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:237)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:62)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:52)
	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more
```

**Expected behavior**
Restart should work seamlessly.

**Actual behavior**
To make it work I have to stop app and start again. Its working until I do a code change and next hot-restart is initiated. 

**Configuration**
```properties
# configure your datasource
quarkus.datasource.db-kind = postgresql
quarkus.datasource.username = postgres
quarkus.datasource.password = pwd
quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/quarkus-03

quarkus.flyway.migrate-at-start=true
```

**Environment (please complete the following information):**
 - Output of `java -version`: 1.8
 - Quarkus version or git rev: 1.11.0.Final

**Comment by gsmet** [comment]
Yes this is a known issue, it's fixed in 1.11.1.Final that we will release tomorrow!

**Comment by geoand** [comment]
Has already been fixed by https://github.com/quarkusio/quarkus/pull/14372
